src/darker/verification.py

```diff
@@ -11,6 +11,50 @@ class NotEquivalentError(Exception):
     pass
 
 
+class BinarySearch:
+    """Effectively search the first index for which a condition is ``True``
+
+    Example usage to find first number between 0 and 100 whose square is larger than
+    1000::
+
+        s = BinarySearch(0, 100)
+        while not s.found:
+            s.respond(s.get_next() ** 2 > 1000)
+        answer = s.result
+
```

Comment:
Maybe make it a doctest with an assertion?
```suggestion
        assert s.result == 32
```